test/integration/node-specific/cursor_async_iterator.test.js

```diff
@@ -79,5 +79,16 @@ describe('Cursor Async Iterator Tests', function () {
 
       expect(count).to.equal(1);
     });
+
+    it('cleans up cursor when breaking out of for await of loops', async function () {
+      const cursor = collection.find();
+
+      for await (const doc of cursor) {
+        expect(doc).to.exist;
+        break;
+      }
+
+      expect(cursor.closed).to.be.true;
+    });
```

Comment:
Let's add just a couple more cases / Qs:
- Won't the finally logic always be run at the end of the loop or is it only when there's break? We can add an assertion to the existing tests for the expected "closed" state either way.
- Can we add a test that attempt to keep using the cursor after the `for await` has done the clean up, ex. `expect(async () => for await (xx)).to.throw(X)`
- We have a gate on calling close only if it has not been called already (using the `closed` flag), I'm not sure it's easily reachable since we have the early return based on `closed=true`. I think it's worth capturing that behavior's significance, does close fail or repeat steps if it's called more than once?
  - We could use the iterator manually, call `cursor[Symbol.asyncIterator]()` and use `.return()` + sinon to check that .close is only called once
  - We could double check that close does not repeat steps if called more than once (this may be more trouble than it's worth in this PR, but I think `needsToEmitClosed` based on `!this[kClosed]` provides idempotency).

Reply:
I've added assertions to the other tests that checks the cursor is closed on every one once they are exahusted. I've also added a test that asserts the second attempted iteration immediately returns, and also spies on the cursor to assert that `close()` is called only once. I think these cover the points you were getting at here but let me know if I missed something.